### PR TITLE
Cyclops change engine state

### PR DESCRIPTION
**Pull request by Jannify**

### CODE REVIEW

File: NitroxClient/Communication/Packets/Processors/CyclopsToggleEngineStateProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+            if (opCyclops.IsPresent())
+            {
+                CyclopsEngineChangeState engineState = opCyclops.Get().GetComponentInChildren<CyclopsEngineChangeState>();
+
+                if (enginePacket.IsOn == engineState.motorMode.engineOn)
+                {
+                    using (packetSender.Suppress<CyclopsEngineChangeState>())
+                    {
+                        engineState.ReflectionSet("startEngine", true);
```
Comment (MarijnS95) [inline NitroxClient/Communication/Packets/Processors/CyclopsToggleEngineStateProcessor.cs]: Is this necessary? `OnClick` will set it to true anyway when the Cyclops if turned off. And when the Cylcops is turned on, this variable is not even read.

Comment (Jannify) [reply NitroxClient/Communication/Packets/Processors/CyclopsToggleEngineStateProcessor.cs]: Yes it is necessary. Prefix() will be set __state to !motor-state and no Packet will send back...

Comment (MarijnS95) [reply NitroxClient/Communication/Packets/Processors/CyclopsToggleEngineStateProcessor.cs]: You're already encapsulating this method in a suppressor.

**Comment by Jannify** [comment]
Limit is not implemented. I don't know how I do that....

**Comment by Jannify** [comment]
Merge Cyclops-MotorMode/State and -HUD to new branch

**Pull request closed without merging**